app/gui/view/debug_scene/component-group/src/lib.rs

```diff
@@ -153,83 +175,74 @@ fn color_component_slider(app: &Application, caption: &str) -> selector::NumberP
+            let radius = 0.px();
+            Circle(radius).fill(color::Rgba::transparent()).into()
+        }
+    }
+}
+
+
 fn init(app: &Application) {
     theme::builtin::dark::register(&app);
     theme::builtin::light::register(&app);
     theme::builtin::light::enable(&app);
 
-    let items_slider = items_slider(app);
     let network = frp::Network::new("Component Group Debug Scene");
-    let selection = create_selection();
-    let selection_animation = Animation::<Vector2>::new(&network);
-    let wide_selection = create_selection();
-    let wide_selection_animation = Animation::<Vector2>::new(&network);
-
-    let component_group = create_component_group(app);
-    component_group.add_child(&selection);
-    let group_name = "Long group name with text overflowing the width";
-    component_group.set_header(group_name.to_string());
-    component_group.set_position_x(-150.0);
-    let dimmed_component_group = create_component_group(app);
-    dimmed_component_group.set_dimmed(true);
-    dimmed_component_group.set_header("Input / Output".to_string());
-    dimmed_component_group.set_position_x(-350.0);
-    let wide_component_group = wide_component_group(app);
-    wide_component_group.add_child(&wide_selection);
 
+    let scroll_area = ScrollArea::new(app);
+    scroll_area.set_position_xy(Vector2(0.0, 100.0));
+    scroll_area.resize(Vector2(170.0, 400.0));
+    scroll_area.set_content_width(150.0);
+    scroll_area.set_content_height(2000.0);
+    app.display.add_child(&scroll_area);
 
-    // === Regular Component Group ===
-
-    frp::extend! { network
-        selection_animation.target <+ component_group.selection_position_target;
-        eval selection_animation.value ((pos) selection.set_position_xy(*pos));
-
-        eval component_group.suggestion_accepted ([](id) DEBUG!("Accepted Suggestion {id}"));
-        eval component_group.expression_accepted ([](id) DEBUG!("Accepted Expression {id}"));
-        eval_ component_group.header_accepted ([] DEBUG!("Accepted Header"));
-    }
-    selection_animation.target.emit(component_group.selection_position_target.value());
-    selection_animation.skip.emit(());
+    let camera = &scroll_area.content_layer().camera();
+    let parent_layer = scroll_area.content_layer();
+    let layers = component_group::Layers::new(&app.logger, camera, parent_layer);
 
+    let group_name = "Long group name with text overflowing the width";
+    let first_component_group = create_component_group(app, group_name, &layers);
+    let group_name = "Second component group";
+    let second_component_group = create_component_group(app, group_name, &layers);
+    second_component_group.set_dimmed(true);
+
+    scroll_area.content().add_child(&first_component_group);
+    scroll_area.content().add_child(&second_component_group);
+
+    // This is a workaround for a bug - without this transparent shape the content of the scroll
+    // area is invisible.
```

Comment:
1. "This" is not defined. Does it refer to a single line?
2. What bug? Such things should always contain a link to a pivotal tracker.
3. This should be a FIXME comment.

Reply:
Fixed